core/src/test/scala/org/apache/spark/SparkContextSuite.scala

```diff
@@ -274,6 +274,31 @@ class SparkContextSuite extends SparkFunSuite with LocalSparkContext {
     }
   }
 
+    test("Default path for file based RDDs is properly set (SPARK-12517)") {
+      sc = new SparkContext(new SparkConf().setAppName("test").setMaster("local"))
+
+      // Test  filetextFile, wholeTextFiles, binaryFiles, hadoopFile and
+      // newAPIHadoopFile for setting the default path as the RDD name
+      val mockPath = "default/path/for/"
+
+      var targetPath = mockPath + "textFile"
+      assert(sc.textFile(targetPath).name == targetPath)
+
```

Comment:
Could you compare by `===` instead of `==` ?
We can get better error messages by `===` operator.